src/action.js

```diff
@@ -145,27 +143,34 @@ inherit(Action, Emitter, {
 
 })
 
+Object.defineProperty(Action.prototype, 'type', {
+  get () {
+    return this.command[this.status]
+  }
+})
+
 /**
  * Generate action methods for each action state
  */
-Object.keys(STATES).forEach(function (key) {
-  const { once, listener } = STATES[key]
+Object.keys(STATES).forEach(function (status) {
+  const { once, disposable, listener } = STATES[status]
 
   /**
    * Create a method to update the action status. For example:
    * action.done({ id: 'earth' })
    */
-  Action.prototype[key] = function (payload) {
+  Action.prototype[status] = function (payload) {
     if (!this.disposable) {
-      this.setStatus(key)
+      this.status = status
+      this.disposable = disposable
```

Comment:
There is duplication here with a`action.setStatus`. I need to think more on this... By being more static here, V8 can optimize these functions way harder.